velox/functions/lib/tests/QuantileDigestTest.cpp

```diff
@@ -952,4 +988,345 @@ TEST_F(QuantileDigestTest, getDistributionFunction) {
+  digest.add(2, 1.0);
+  digest.add(3, 1.0);
+
+  EXPECT_NEAR(getTruncatedMean(digest, 0.0, 1.0).value(), 2.0, 1e-6);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.5, 1.0).value(), 4.0 / 1.5, 1e-6);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.0, 0.5).value(), 2.0 / 1.5, 1e-6);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.0, 0.1).value(), 1.0, 1e-6);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.9, 0.99).value(), 3.0, 1e-6);
+}
+
+TEST_F(QuantileDigestTest, truncatedMeanSkewed) {
+  QuantileDigest<int64_t> digest{StlAllocator<int64_t>(allocator()), 0.01};
+  digest.add(1, 1.0);
+  digest.add(5, 1.0);
+  digest.add(10, 1.0);
+  digest.add(20, 1.0);
+  digest.add(50, 1.0);
+
+  EXPECT_NEAR(getTruncatedMean(digest, 0.25, 0.75).value(), 11.5, 0.1);
+}
+
+TEST_F(QuantileDigestTest, truncatedMeanRepeated) {
+  QuantileDigest<int64_t> digest{StlAllocator<int64_t>(allocator()), 0.01};
+  digest.add(1, 1.0);
+  for (int i = 0; i < 8; i++) {
+    digest.add(2, 1.0);
+  }
+  digest.add(5, 1.0);
+
+  EXPECT_NEAR(getTruncatedMean(digest, 0.0, 0.5).value(), 1.8, 0.1);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.4, 0.6).value(), 2.0, 0.01);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.8, 1.0).value(), 3.5, 0.1);
+}
+
+TEST_F(QuantileDigestTest, truncatedMeanSigned) {
+  QuantileDigest<int64_t> digest{StlAllocator<int64_t>(allocator()), 0.01};
+  digest.add(-2, 1.0);
+  digest.add(-1, 1.0);
+  digest.add(0, 1.0);
+  digest.add(1, 1.0);
+  digest.add(2, 1.0);
+
+  EXPECT_NEAR(getTruncatedMean(digest, 0.0, 0.4).value(), -1.5, 0.1);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.4, 0.6).value(), 0.0, 0.01);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.2, 0.8).value(), 0.0, 0.01);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.4, 1.0).value(), 1.0, 0.1);
+}
+
+TEST_F(QuantileDigestTest, truncatedMeanDoubles) {
+  QuantileDigest<double> digest{StlAllocator<double>(allocator()), 0.01};
+  digest.add(0.2, 1.0);
+  digest.add(0.8, 1.0);
+  digest.add(1.4, 1.0);
+  digest.add(2.0, 1.0);
+  digest.add(2.6, 1.0);
+
+  EXPECT_NEAR(getTruncatedMean(digest, 0.0, 1.0).value(), 1.4, 0.01);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.6, 1.0).value(), 2.3, 0.1);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.0, 0.4).value(), 0.5, 0.1);
+}
+
+TEST_F(QuantileDigestTest, truncatedMeanWeighted) {
+  QuantileDigest<int64_t> digest{StlAllocator<int64_t>(allocator()), 0.01};
+  digest.add(2, 1.0);
+  digest.add(4, 2.0);
+  digest.add(5, 3.0);
+  digest.add(6, 3.0);
+  digest.add(10, 1.0);
+
+  EXPECT_NEAR(getTruncatedMean(digest, 0.0, 0.1).value(), 2.0, 0.01);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.2, 0.8).value(), 31.0 / 6.0, 0.01);
+  EXPECT_NEAR(getTruncatedMean(digest, 0.6, 1.0).value(), 28.0 / 4.0, 0.01);
+}
+
+TEST_F(QuantileDigestTest, truncatedMeanBoundaries) {
+  QuantileDigest<int64_t> digest{StlAllocator<int64_t>(allocator()), 0.01};
+  for (int64_t i = 1; i <= 10; i++) {
+    digest.add(i, 1.0);
+  }
+
+  // Test exact boundaries
+  EXPECT_NEAR(getTruncatedMean(digest, 0.0, 1.0).value(), 5.5, 0.01);
```

Comment:
nit: Why use 0.01 as the error bound here while using 0.1 at other checks? Could we use the same error bound at all checks (except the ones that use 1e-10 and 1e5 for very large and very small values)?